geopandas/geodataframe.py

```diff
@@ -2025,45 +2025,41 @@ def sjoin(self, df, *args, **kwargs):
-        3       Lobamba  POINT (31.20000 -26.46667)
-        4    Luxembourg    POINT (6.13000 49.61166)
-
-        >>> cities_w_country_data = cities.sjoin(countries)
-        >>> cities_w_country_data.head()  # doctest: +SKIP
-                name_left                   geometry  index_right   pop_est \
-    continent name_right iso_a3  gdp_md_est
-        0    Vatican City  POINT (12.45339 41.90328)          141  62137802 \
-    Europe    Italy    ITA   2221000.0
-        1    San Marino  POINT (12.44177 43.93610)          141  62137802 \
-    Europe    Italy    ITA   2221000.0
-        192          Rome  POINT (12.48131 41.89790)          141  62137802 \
-    Europe    Italy    ITA   2221000.0
-        2           Vaduz   POINT (9.51667 47.13372)          114   8754413 \
-    Europe    Au    stria    AUT    416600.0
-        184        Vienna  POINT (16.36469 48.20196)          114   8754413 \
-    Europe    Austria    AUT    416600.0
+        >>> import geodatasets
+        >>> chicago = geopandas.read_file(
+        ...     geodatasets.get_path("geoda.chicago_health")
+        ... )
+        >>> groceries = geopandas.read_file(
+        ...     geodatasets.get_path("geoda.groceries")
+        ... ).to_crs(chicago.crs)
+
+        >>> chicago.head()  # doctest: +SKIP
+           ComAreaID  ...                                           geometry
+        0         35  ...  POLYGON ((-87.60914 41.84469, -87.60915 41.844...
+        1         36  ...  POLYGON ((-87.59215 41.81693, -87.59231 41.816...
+        2         37  ...  POLYGON ((-87.62880 41.80189, -87.62879 41.801...
+        3         38  ...  POLYGON ((-87.60671 41.81681, -87.60670 41.816...
+        4         39  ...  POLYGON ((-87.59215 41.81693, -87.59215 41.816...
+        [5 rows x 87 columns]
+
+        >>> groceries.head()  # doctest: +SKIP
+           OBJECTID     Ycoord  ...  Category                         geometry
+        0        16  41.973266  ...       NaN  MULTIPOINT (-87.65661 41.97321)
+        1        18  41.696367  ...       NaN  MULTIPOINT (-87.68136 41.69713)
+        2        22  41.868634  ...       NaN  MULTIPOINT (-87.63918 41.86847)
+        3        23  41.877590  ...       new  MULTIPOINT (-87.65495 41.87783)
+        4        27  41.737696  ...       NaN  MULTIPOINT (-87.62715 41.73623)
+        [5 rows x 8 columns]
+
+        >>> groceries_w_communities = groceries.sjoin(chicago)
+        >>> groceries_w_communities.head()  # doctest: +SKIP
+             OBJECTID     Ycoord     Xcoord  ... GonorrF GonorrM Tuberc
+        0          16  41.973266 -87.657073  ...   170.8   468.7   13.6
+        87        365  41.961707 -87.654058  ...   170.8   468.7   13.6
+        90        373  41.963131 -87.656352  ...   170.8   468.7   13.6
+        140       582  41.969131 -87.674882  ...   170.8   468.7   13.6
+        1          18  41.696367 -87.681315  ...   800.5   741.1    2.6
+        [5 rows x 95 columns]
```

Comment:
I don't find this a very illustrating example (also because of the clipping of the repr). For example, you don't even see that there are still geometries in the output (and that it are the points). 

I am not sure if we have good other datasets in geodatasets at the moment (or create some small dummy data to illustrate?), but maybe making the output a bit wider (as was done before as well with the `\`) might already help. It would also help if it were POINTs instead of MULTIPOINTs ..

Reply:
Or selecting some columns as you did below for nearest